### PR TITLE
FAI-5208 - Fix incremental logic of CircleCI connector

### PR DESCRIPTION
**Pull request by cjwooo**
## Description

Convert pipelines stream to full_refresh only. Pipelines are like GitHub repositories, in that while they do have an "update" timestamp, they are not directly updated often, and so incremental pulling of pipelines will likely lead to unintended skipping of data. Workflows have been separated into a separate incremental stream. This also minimizes the likelihood of Airbyte blowing up due to overly large records, as each pipeline record used to include every single workflow and job for that pipeline.

## Type of change
- [x] Bug fix
- [x] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

File: destinations/airbyte-faros-destination/src/converters/circleci/pipelines.ts [hunk 1]
```diff
@@ -1,74 +1,9 @@
...
-            type: CircleCICommon.convertJobType(job.type),
-            build: buildKey,
-          },
-        });
-      }
-    }
-    return res;
+  readonly destinationModels: ReadonlyArray<DestinationModel> = [];
+  async convert(): Promise<ReadonlyArray<DestinationRecord>> {
+    return [];
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/converters/circleci/pipelines.ts:7]: nothing to write here?

File: sources/circleci-source/src/streams/common.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import {AirbyteLogger, AirbyteStreamBase} from 'faros-airbyte-cdk';
+
+import {CircleCI, CircleCIConfig} from '../circleci/circleci';
+
+export abstract class StreamBase extends AirbyteStreamBase {
```
Comment (tovbinm) [inline sources/circleci-source/src/streams/common.ts:5]: perhaps rename to `CircleCIStreamBase`

**Review by tovbinm — APPROVED** [review]

File: destinations/airbyte-faros-destination/src/converters/circleci/pipelines.ts (hunk 1 above)
Comment (cjwooo) [reply destinations/airbyte-faros-destination/src/converters/circleci/pipelines.ts:7]: Correct.

File: sources/circleci-source/src/streams/workflows.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  ): StreamState {
+    const pipelineId = latestRecord.pipeline_id;
+    const lastStoppedAt = currentStreamState[pipelineId]?.lastStoppedAt;
+    if (
+      latestRecord.stopped_at &&
+      new Date(latestRecord.stopped_at) > new Date(lastStoppedAt ?? 0)
+    ) {
+      return {
+        ...currentStreamState,
+        [pipelineId]: {lastStoppedAt: latestRecord.stopped_at},
```
[the remark this answers is not in the document]
Comment (cjwooo) [reply sources/circleci-source/src/streams/workflows.ts:65]: References `pipelineId` variable as an object key, as opposed to the string literal `"pipelineId"`

**Pull request merged**